physionet-django/console/test_views.py

```diff
@@ -614,3 +615,181 @@ def test_frontpage_button_delete(self):
+
+    def test_event_agreement_new_version_valid(self):
+        """tests the view that adds a valid new version of event agreement"""
+
+        event_agreement = self.test_add_event_agreement_valid()
+
+        # Create an event Agreement
+        response = self.client.post(
+            reverse('event_agreement_new_version', args=[event_agreement.pk]),
+            data={
+                'name': self.event_agreement_name,
+                'version': self.event_agreement_version_new_version,
+                'slug': self.event_agreement_slug_new_version,
+                'is_active': True,
+                'html_content': self.event_agreement_html_content,
+                'access_template': self.event_agreement_access_template
+            })
+
+        self.assertEqual(response.status_code, 302)
+        event_agreement = EventAgreement.objects.get(slug=self.event_agreement_slug_new_version)
+        self.assertEqual(event_agreement.version, self.event_agreement_version_new_version)
+
+    def test_event_agreement_new_version_invalid_slug(self):
+        """tests the view that adds an invalid new version of event agreement(invalid slug)"""
+
+        event_agreement = self.test_add_event_agreement_valid()
+
+        # Create an event Agreement
+        response = self.client.post(
+            reverse('event_agreement_new_version', args=[event_agreement.pk]),
+            data={
+                'name': self.event_agreement_name,
+                'version': self.event_agreement_version_new_version,
+                'slug': self.event_agreement_slug,
+                'is_active': True,
+                'html_content': self.event_agreement_html_content,
+                'access_template': self.event_agreement_access_template
+            })
+
+        self.assertEqual(response.status_code, 200)
+        self.assertTemplateUsed(response, 'console/event_agreement_new_version.html')
+        self.assertContains(response, 'Event agreement with this Slug already exists.')
+
+    def test_event_agreement_new_version_invalid_version(self):
+        """tests the view that adds an invalid new version of event agreement(invalid version)"""
+
+        event_agreement = self.test_add_event_agreement_valid()
+
+        # Create an event Agreement
+        response = self.client.post(
+            reverse('event_agreement_new_version', args=[event_agreement.pk]),
+            data={
+                'name': self.event_agreement_name,
+                'version': self.event_agreement_version_invalid,
+                'slug': self.event_agreement_slug_new_version,
+                'is_active': True,
+                'html_content': self.event_agreement_html_content,
+                'access_template': self.event_agreement_access_template
+            })
+
+        self.assertEqual(response.status_code, 200)
+        self.assertTemplateUsed(response, 'console/event_agreement_new_version.html')
+        self.assertContains(response,
```

Comment:
I don't think asserting the response contains specific text is necessary. This just means that in the future if we want to update the text we have to also update the test. Asserting the server error is caught is sufficient.

Reply:
Updated this, Quick note in some cases because of how views are implemented the status code will be 200 and in most cases it will be 300.

So in terms of readability, it might look like we created a valid Event Agreement, and tried to create and invalid eventagreement or new invalid event agreement. but didnot do further verification except the status code and template(but the combination of status code and template used seems sufficient here)